homeassistant/components/yolink/binary_sensor.py

```diff
@@ -132,7 +132,13 @@ def __init__(
     @callback
     def update_entity_state(self, state: dict[str, Any]) -> None:
         """Update HA Entity State."""
+        self._attr_available = self.coordinator.dev_online
         self._attr_is_on = self.entity_description.value(
             state.get(self.entity_description.state_key)
         )
         self.async_write_ha_state()
+
+    @property
+    def available(self):
+        """Return true is device is available."""
+        return self.coordinator.dev_online
```

Comment:
We should also check `super().available` since the coordinator entity already defines that.

https://github.com/home-assistant/core/blob/067b94899f2f77a5ede4f661363bd71942bb3c9f/homeassistant/helpers/update_coordinator.py#L493-L496

Reply:
Please address this comment.